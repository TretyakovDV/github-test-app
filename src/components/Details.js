import React from 'react';
import { observer, inject } from 'mobx-react';
import {
  Paper, IconButton, Typography, Chip,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import AccountTreeIcon from '@material-ui/icons/AccountTree';
import StarIcon from '@material-ui/icons/Star';
import VisibilityIcon from '@material-ui/icons/Visibility';

const Details = inject('details', 'repositories')(observer(({ details, repositories }) => {
  const handleClose = () => details.close();
  const repository = repositories.getRepository(details.id);
  const languages = repositories.languages(details.id);

  if (!details.isOpen) return null;

  return (
    <Paper elevation={10} className="details">
      <div>
        <IconButton onClick={handleClose}>
          <CloseIcon />
        </IconButton>
      </div>
      <Typography variant="h2" className="mt-40 mb-20">{repository.title}</Typography>
      <Typography variant="h4" className="mt-40 mb-20">Stats</Typography>
      <Typography variant="h5" className="details__icon-wrapper">
        <StarIcon className="details__icon" />
        Stars:
        {' '}
        {repository.stars}
      </Typography>
      <Typography variant="h5" className="details__icon-wrapper">
        <VisibilityIcon className="details__icon" />
        Watchers:
        {' '}
        {repository.watchers}
      </Typography>
      <Typography variant="h5" className="details__icon-wrapper">
        <AccountTreeIcon className="details__icon" />
        Forks:
        {' '}
        {repository.forks}
      </Typography>
      <Typography variant="h4" className="mt-40 mb-20">Languages</Typography>
      <div className="languages-container">
        {languages.map((lang) => (
          <Chip key={lang.label} label={`${lang.label} - ${Number(lang.percent).toFixed(4)}%`} className="languages__item" />
        ))}
      </div>
    </Paper>
  );
}));

export default Details;
