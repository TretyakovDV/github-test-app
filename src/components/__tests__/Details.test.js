import React from 'react';
import { fireEvent } from '@testing-library/react';
import renderWithProvider from '../../utils/test/renderWithProvider';
import Details from '../Details';

describe('Details', () => {
  const close = jest.fn();
  const defaultStore = {
    details: {
      isOpen: true,
      close,
    },
    repositories: {
      getRepository: jest.fn(() => ({
        title: 'title',
        stars: 1,
        forks: 1,
        watchers: 1,
      })),
      languages: jest.fn(() => ([{ label: 'label', percent: 100 }])),
    },
  };

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<Details />, defaultStore);
    expect(asFragment()).toMatchSnapshot();
  });

  it('when isOpen is false', () => {
    const store = {
      ...defaultStore,
      details: {
        isOpen: false,
      },
    };

    const { asFragment } = renderWithProvider(<Details />, store);
    expect(asFragment()).toMatchSnapshot();
  });

  it('close', () => {
    const { getByTestId } = renderWithProvider(<Details />, defaultStore);
    fireEvent.click(getByTestId('close'));
    expect(close).toHaveBeenCalledTimes(1);
  });
});
