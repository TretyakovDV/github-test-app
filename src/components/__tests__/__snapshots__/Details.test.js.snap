// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root details MuiPaper-elevation10 MuiPaper-rounded"
  >
    <div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <h2
      class="MuiTypography-root mt-40 mb-20 MuiTypography-h2"
    >
      title
    </h2>
    <h4
      class="MuiTypography-root mt-40 mb-20 MuiTypography-h4"
    >
      Stats
    </h4>
    <h5
      class="MuiTypography-root details__icon-wrapper MuiTypography-h5"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root details__icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      Stars: 1
    </h5>
    <h5
      class="MuiTypography-root details__icon-wrapper MuiTypography-h5"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root details__icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
        />
      </svg>
      Watchers: 1
    </h5>
    <h5
      class="MuiTypography-root details__icon-wrapper MuiTypography-h5"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root details__icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z"
        />
      </svg>
      Forks: 1
    </h5>
    <h4
      class="MuiTypography-root mt-40 mb-20 MuiTypography-h4"
    >
      Languages
    </h4>
    <div
      class="languages-container"
    >
      <div
        class="MuiChip-root languages__item"
      >
        <span
          class="MuiChip-label"
        >
          label - 100.0000%
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Details when isOpen is false 1`] = `<DocumentFragment />`;
